src/prefect/events/worker.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+import atexit
+import queue
+import threading
+from contextlib import asynccontextmanager, contextmanager
+from typing import Any, Dict, Optional, Type
+
+from prefect._internal.compatibility.experimental import experiment_enabled
+from prefect.settings import PREFECT_API_KEY, PREFECT_API_URL, PREFECT_CLOUD_API_URL
+
+from .clients import EventsClient, NullEventsClient, PrefectCloudEventsClient
+from .schemas import Event
+
+
+class EventsWorker:
+    def __init__(
+        self,
+        client_type: Type[EventsClient],
+        client_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        self._client_type = client_type
+        self._client_kwargs = client_kwargs or {}
+        self._client: EventsClient
+
+        self._queue: queue.SimpleQueue[Event] = queue.SimpleQueue()
+
+        self._thread = threading.Thread(
+            target=self._start_loop, name="events-worker", daemon=True
+        )
+        self._stop_event = threading.Event()
+        self._ready_event = threading.Event()
+        self._lock = threading.Lock()
+
+        atexit.register(self.stop)
+
+    def _start_loop(self):
+        """Entrypoint for the thread."""
+        loop = asyncio.new_event_loop()
+        loop.run_until_complete(self._main_loop())
+        loop.close()
+
+    async def _main_loop(self):
+        """Orchestrate the events client and emit any queued events."""
+        self._client = self._client_type(**self._client_kwargs)
+        async with self._client:
+            self._ready_event.set()
```

Comment:
You'll want to use a future for this, so you can raise an exception if there's a failure starting the worker thread. I would recommend reviewing the implementation in https://github.com/PrefectHQ/prefect/blob/main/src/prefect/_internal/concurrency/runtime.py#L44-L74